### PR TITLE
Allow k3s to customize apiServerPort on helm-controller

### PR DESCRIPTION
**Pull request by StrongMonkey**
#### Proposed Changes ####

When using helm-controller with bootstrap option, the apiServer port is hard-coded to 6443. This means that if you are using a port other than 6443 for api-server, it will be broken. Instead, it should be parameterized and pass down apiServerPort config to helm controller 

Helm-controller PR: https://github.com/k3s-io/helm-controller/pull/198

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/7835

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,6 +16,8 @@ replace (
 	github.com/golang/protobuf => github.com/golang/protobuf v1.5.3
 	github.com/googleapis/gax-go/v2 => github.com/googleapis/gax-go/v2 v2.1.1
 	github.com/juju/errors => github.com/k3s-io/nocode v0.0.0-20200630202308-cb097102c09f
+
+	github.com/k3s-io/helm-controller => github.com/acorn-io/helm-controller v0.0.0-20230626182853-bb03cf954008
```
Comment (dereknola) [inline go.mod:20]: Why are we now replacing a k3s-io org repo with an external one?

Comment (brandond) [reply go.mod:20]: I believe this is just to test the change before  https://github.com/k3s-io/helm-controller/pull/198 is merged.

Comment (StrongMonkey) [reply go.mod:20]: This will be changed once the dependency PR is merged

Comment (brandond) [reply go.mod:20]: https://github.com/k3s-io/helm-controller/releases/tag/v0.15.2

**Review by brandond — APPROVED** [review]

**Comment by brandond** [comment]
LGTM, looks like you need to rebase and resolve a conflict though.

**Comment by StrongMonkey** [comment]
@brandond thanks for the quick approval, rebased 👍

**Review by dereknola — APPROVED** [review]

**Pull request merged**